test/Microsoft.Framework.DependencyInjection.Tests/ServiceCollectionExtensionTest.cs

```diff
@@ -276,87 +285,154 @@ public void TryAddWithEnumerableReturnsFalseIfNoneAdded()
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(TryAddEnumerableImplementationTypeData))]
+        public void TryAddEnumerable_AddsService(
+            ServiceDescriptor descriptor,
+            Type expectedServiceType,
+            Type expectedImplementationType,
+            ServiceLifetime expectedLifetime)
         {
             // Arrange
             var collection = new ServiceCollection();
-            collection.TryAddEnumerable(ServiceDescriptor.Transient<IFakeService, FakeService>());
 
             // Act
-            var result = collection.TryAddEnumerable(ServiceDescriptor.Transient<IFakeService, FakeService>());
+            var result = collection.TryAddEnumerable(descriptor);
 
             // Assert
-            Assert.False(result);
+            Assert.True(result);
 
-            var descriptor = Assert.Single(collection);
-            Assert.Equal(typeof(IFakeService), descriptor.ServiceType);
-            Assert.Equal(typeof(FakeService), descriptor.ImplementationType);
-            Assert.Equal(ServiceLifetime.Transient, descriptor.Lifetime);
+            var d = Assert.Single(collection);
+            Assert.Equal(expectedServiceType, d.ServiceType);
+            Assert.Equal(expectedImplementationType, d.GetImplementationType());
+            Assert.Equal(expectedLifetime, d.Lifetime);
         }
 
-        [Fact]
-        public void TryAddEnumerable_AllowsMultipleRegistrations()
+
+        [Theory]
+        [MemberData(nameof(TryAddEnumerableImplementationTypeData))]
+        public void TryAddEnumerable_DoesNotAddDuplicate(
+            ServiceDescriptor descriptor,
+            Type expectedServiceType,
+            Type expectedImplementationType,
+            ServiceLifetime expectedLifetime)
         {
             // Arrange
             var collection = new ServiceCollection();
-            collection.TryAddEnumerable(ServiceDescriptor.Transient<IFakeService, FakeService>());
+            collection.TryAddEnumerable(descriptor);
 
             // Act
-            var result = collection.TryAddEnumerable(ServiceDescriptor.Transient<IFakeService, IFakeService>());
+            var result = collection.TryAddEnumerable(descriptor);
 
             // Assert
-            Assert.True(result);
+            Assert.False(result);
 
-            Assert.Single(collection, sd => sd.ImplementationType == typeof(FakeService));
-            Assert.Single(collection, sd => sd.ImplementationType == typeof(IFakeService));
+            var d = Assert.Single(collection);
+            Assert.Equal(expectedServiceType, d.ServiceType);
+            Assert.Equal(expectedImplementationType, d.GetImplementationType());
+            Assert.Equal(expectedLifetime, d.Lifetime);
         }
 
         [Fact]
-        public void TryAddEnumerable_AllowsMultipleRegistrations_WithFactory()
+        public void TryAddEnumerable_WithEnumerable_ReturnsTrueIfAnyAdded()
```

Comment:
I was going for consistency with the existing `TryAdd` implementation. For the EF use case, the calling code ignores the return value either way.